docs/website/blog/deploy-and-connect-to-microservice-using-helm-and-exec.md

```diff
@@ -0,0 +1,815 @@
+```
+</details>
+
+## 4. Modify the Devfile
+
+We will be using `odo deploy` for deploying our application, and for this we need to modify the Devfile by adding required commands and components.
+
+### Add the `commands`
+Let us begin by first adding a `deploy` command under the `commands` section.
+```yaml
+# This is the main "composite" command that will run all below commands
+- id: deploy
+  composite:
+    commands:
+      - k8s-serviceaccount-for-helm
+      - k8s-role-for-helm
+      - k8s-rolebinding-for-helm
+      - deploy-db
+      - build-image
+      - k8s-deployment
+      - k8s-service
+      - k8s-url
+    group:
+      isDefault: true
+      kind: deploy
+```
+
+`deploy` command is a composition of various other commands in the order in which we want them to be executed.
+For e.g. before deploying the database with helm, we need to ensure a service account with the required permissions (made possible by role and rolebinding) has been created;
+and so we run `k8s-serviceaccount-for-helm`, `k8s-role-for-helm` and `k8s-rolebinding-for-helm` before running `deploy-db` command.
+
+Let us now add the individual commands.
+
+We will first define the `deploy-db` command that is used to deploy the helm chart.
+
+To use an external tool such as helm or kustomize, we need to ensure 2 things:
+1. use an `exec` command; learn more [here](/docs/development/devfile#how-odo-runs-exec-commandsin-deploy-mode).
+2. the container component referenced by this command uses an image that contains the required binary.
+
+```yaml
+- id: deploy-db
+  exec:
+    commandLine: helm repo add bitnami https://charts.bitnami.com/bitnami && helm repo update && helm install mongodb bitnami/mongodb
+    component: deploy-db
+```
+
+We will now add the remaining commands.
+
+```yaml
+- id: k8s-serviceaccount-for-helm
+  apply:
+    component: outerloop-serviceaccount
+- id: k8s-role-for-helm
+  apply:
+    component: outerloop-role
+- id: k8s-rolebinding-for-helm
+  apply:
+    component: outerloop-rolebinding
+- id: build-image
+  apply:
+    component: outerloop-build
+- id: k8s-deployment
+  apply:
+    component: outerloop-deployment
+- id: k8s-service
+  apply:
+    component: outerloop-service
+- id: k8s-url
+  apply:
+      component: outerloop-url
+```
+
+### Add the `components`
+Every command above references a `component`, and so we now add components under the `components` section.
+
+We will first add the component referenced by `deploy-db` command.
+
+```yaml
+- name: deploy-db
+  container:
+    image: quay.io/tkral/devbox-demo-devbox
+  attributes:
+    pod-overrides:
+      spec:
+        serviceAccountName: my-go-app
+```
+
+The image used by this container component contains the Helm binary that we can use to deploy the helm chart.
+
+The component is using a `pod-overrides` attribute that will override the service account used by the pod to deploy the helm chart to use the service account (`my-go-app`) we define in this Devfile.
```

Comment:
Can you link to the `pod-overrides` documentation to learn more: https://devfile.io/docs/2.2.0/overriding-pod-and-container-attributes#pod-overrides